src/pylint_datetime/__init__.py

```diff
@@ -1,4 +1,185 @@
-# License: MIT
-# Copyright © 2023 Frequenz Energy-as-a-Service GmbH
+"""Pylint checker ensure correct usage of datetime module, especially around aware/naive objects."""
 
-"""checker to ensure correct datetime usage with aware objects in datetime module."""
+from typing import Any
+
+import astroid  # type: ignore
+import pylint  # type: ignore
+from pylint.checkers import BaseChecker  # type: ignore
+
+# function calls to fromisoformat, strptime and fromisoformat can't be checked - they parse strings
+# if the specified timezone is None, still allowed although they produce naive objects
+
+
+class DatetimeChecker(BaseChecker):  # type: ignore
+    """class for custom pylint checker."""
+
+    name = "pylint-datetime"
+    priority = -1
+    msgs = {
+        "W9999": (
+            "timedelta() called without keyword arguments",
+            "timedelta-no-keyword-args",
+            "timedelta() called without keyword arguments.",
+        ),
+        "W9998": (
+            'Function call to "%s"/"replace" should be called with a timezone argument',
+            "timezone-no-argument",
+            'Function call to "%s"/"replace" should be called with a timezone argument',
+        ),
+        "W9997": (
+            'Function call to "%s" can only produce naive objects',
+            "naive-datetime-call",
+            'Function call to "%s" can only produce naive objects, aware objects necessary',
+        ),
+        "W9996": (
+            'Attribute access "%s" should be followed by a call to replace with timezone argument',
+            "naive-datetime-no-timezone",
+            'Attribute access "%s" should be followed by a call to replace with timezone argument',
+        ),
+    }
+
+    def __init__(self, linter: pylint.lint.PyLinter = None) -> None:
+        """Initialize checker.
+
+        Args:
+            linter: instance of pylint linter.
+        """
+        super().__init__(linter)
+
+    def timedelta_with_keywords(
+        self, node: astroid.node_classes.NodeNG, name: str
+    ) -> None:
+        """Check calls to timedelta() and datetime.timedelta() for keyword arguments.
+
+        Args:
+            node: the node that visit_call originates from.
+            name: name of the function being called.
+        """
+        if name == "timedelta":
+            for _ in node.args:
+                self.add_message("timedelta-no-keyword-args", node=node)
+            if not node.args and not node.keywords:
+                self.add_message("timedelta-no-keyword-args", node=node)
+
+    # check for positional arguments cheesy, uses fact, that they have one Attribute Argument.
```

Comment:
`cheesy`? What is `fact`? Shouldn't this be part of the method documentation instead of a separate comment?

Reply:
What I meant is that it is pretty difficult to check if the timezone was passed as a positional argument (because essentially any value is accepted as a timezone) and right now the check only works because all the functions that get checked have exactly one argument of type Attribute. If say a function gets added to the list that has multiple the check wouldn't work.

Reply:
I guess it would be less cheesy if I also checked the type of the argument